Branch-SDK/src/main/java/io/branch/referral/util/BranchCPID.java

```diff
@@ -0,0 +1,101 @@
+package io.branch.referral.util;
+
+import android.text.TextUtils;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+public class BranchCPID {
+
+    JSONObject cpidData;
+
+    public BranchCPID() {
```

Comment:
Needs a unit test.

Reply:
None of the code related to CPID was changed, just refactored into new classes. There is already a test called `testGetCPID`, which is not very exhaustive, so added a new test `testGetCPID` in the TestBed module coz those tests mimic a running application with real Branch keys and all, which is needed for the CPID request to succeed.